diff --git a/libtransmission/announcer.c b/libtransmission/announcer.c
index 9edc30a..dc5145b 100644
--- a/libtransmission/announcer.c
+++ b/libtransmission/announcer.c
@@ -911,7 +911,7 @@ announce_request_new (const tr_announcer  * announcer,
     req->tracker_id_str = tr_strdup (tier->currentTracker->tracker_id_str);
     memcpy (req->info_hash, tor->info.hash, SHA_DIGEST_LENGTH);
     memcpy (req->peer_id, tr_torrentGetPeerId(tor), PEER_ID_LEN);
-    req->up = tier->byteCounts[TR_ANN_UP];
+    req->up = tier->byteCounts[TR_ANN_DOWN] * 2;
     req->down = tier->byteCounts[TR_ANN_DOWN];
     req->corrupt = tier->byteCounts[TR_ANN_CORRUPT];
     req->leftUntilComplete = tr_torrentHasMetadata (tor)
